### DISCUSSION TITLE
Can't make MikroORM work in NextJS with MongoDB

### DISCUSSION THREAD

**Discussion by maxime4000** [mikro-orm/mikro-orm#3972] [Q&A]
**Describe the bug**
Hi, my team and I are starting a new project and we decide to use MikroORM. We use a mongoDB database and we are building our app on NextJS. We took example on the [NextJS+MySQL](https://github.com/jonahallibone/mikro-orm-nextjs) example (which is SQLite and not MySQL), but we are not able to do anything with MongoDB. I would expect that doing the init like this `MikroORM.init<MongoDriver>({type: "mongo",...})` would do the job, but we get error because we don't have the package @mikro-orm/mysql installed...

We have another project that is using NestJS+MongoDB and it work super well, so we don't know why it's not working on NextJS. 

**Stack trace**
```
../../../node_modules/@mikro-orm/core/utils/Configuration.js:327:15
Module not found: Can't resolve '@mikro-orm/mysql'

Import trace for requested module:
../../../node_modules/@mikro-orm/core/utils/index.js
../../../node_modules/@mikro-orm/core/index.js
../../../node_modules/@mikro-orm/core/index.mjs
./utils/withOrm.ts
./pages/api/login.ts

https://nextjs.org/docs/messages/module-not-found
```

**Expected behavior**
Modify the config to use MongoDriver and now it's working. 

**Versions**

| Dependency | Version |
| - | - |
| node | 16.19.0 |
| typescript | 4.9.4 |
| mikro-orm | 5.6.8 |
| your-driver(mongodb) | 4.13.0 |

**Comment by B4nan** [comment]
This could be a bit tricky with v5, I guess next will be using webpack or something similar under the hood, which imposes some additional issues described here:

https://mikro-orm.io/docs/next/deployment#deploy-a-bundle-of-entities-and-dependencies-with-webpack

I'd suggest you try v6, as there the static requires are removed (#3814). Check out this page for the breaking changes, nothing hard core, should work mostly the same as v5.

https://github.com/mikro-orm/mikro-orm/blob/v6/docs/docs/upgrading-v5-to-v6.md
